receiver/nginxreceiver/scraper_test.go

```diff
@@ -133,8 +138,273 @@ func TestScraperError(t *testing.T) {
 	})
 }
 
+func TestScraperLogs(t *testing.T) {
+	l, err := net.Listen("tcp", "127.0.0.1:52418")
```

Comment:
Generally, we need to be more flexible when assigning a port. You can let the system find one that is available: https://stackoverflow.com/a/43425461

(Integration tests should probably do this as well, but that's a bit trickier. We can at least be flexible in unit tests)

Reply:
fixed for unit test